hal/src/main/native/athena/HAL.cpp

```diff
@@ -285,8 +288,52 @@ size_t HAL_GetSerialNumber(char* buffer, size_t size) {
 }
 
 size_t HAL_GetComments(char* buffer, size_t size) {
-  // TODO: implement file parsing
-  return 0;
+  fs::path path = "/etc/machine-info";
+  std::error_code code;
+  std::unique_ptr<wpi::MemoryBuffer> fileBuffer =
+      wpi::MemoryBuffer::GetFile(path.string(), code);
+
+  std::string fileContents;
+  if (fileBuffer && !code) {
+    // If the buffer is valid put the file into a string
+    fileContents = std::string_view(
+        reinterpret_cast<const char*>(fileBuffer->begin()), fileBuffer->size());
+  } else {
+    // Otherwise null terminate if there's room and return
+    if (size > 0) {
+      buffer[0] = '\0';
+    }
+    return 0;
+  }
+
+  std::string_view searchString = "PRETTY_HOSTNAME=\"";
+
+  if (!wpi::contains(fileContents, searchString)) {
+    // If the file doesn't contain the data null terminate if there's room and
+    // return
+    if (size > 0) {
+      buffer[0] = '\0';
+    }
+    return 0;
+  }
+
+  size_t start = fileContents.find(searchString) + searchString.size();
+  size_t end = fileContents.find("\"", start);
```

Comment:
This won't handle escaped double-quotes, e.g. `\"`, and also doesn't handle `\\`.  To do this right, we should do a character-by-character scan and handle escapes individually.  I'm pretty sure we have this code somewhere else already for C-style strings, let me do some searching.

Reply:
Found it.  We could hoist this to wpiutil. https://github.com/wpilibsuite/allwpilib/blob/323524fed62053552402a2631241fb0ae609e683/glass/src/libnt/native/cpp/NetworkTables.cpp#L841-L877